hal/halx86/generic/reboot.c

```diff
@@ -29,22 +29,13 @@ VOID
 NTAPI
 HalpReboot(VOID)
 {
+    PHYSICAL_ADDRESS PhysicalAddress;
     UCHAR Data;
     PVOID ZeroPageMapping;
-    PHARDWARE_PTE Pte;
 
-    /* Get a PTE in the HAL reserved region */
-    ZeroPageMapping = (PVOID)(0xFFC00000 + PAGE_SIZE);
-    Pte = GetPteAddress(ZeroPageMapping);
-
-    /* Make it valid and map it to the first physical page */
-    Pte->Valid = 1;
-    Pte->Write = 1;
-    Pte->Owner = 1;
-    Pte->PageFrameNumber = 0;
-
-    /* Flush the TLB by resetting CR3 */
-    __writecr3(__readcr3());
+    /* Map the first physical page */
+    PhysicalAddress.QuadPart = 0;
+    ZeroPageMapping = HalpMapPhysicalMemory64(PhysicalAddress, 1);
```

Comment:
Maybe you can remove the GetPteAddress macro line 16 altogether.

Reply:
It should also be noted that the HalpMapPhysicalMemory64 function seems to set (at least directly) the 'Owner' flag of the Pte. No idea whether it's mandatory or not.

Reply:
Where do you see Owner being set?

Reply:
ERRR... the comment should read: "HalpMapPhysicalMemory64 function seems to **not** set". Sorry for the typo. Yes it doesn't set the flag, while the original code does, hence I was wondering whether it is intentional.

Reply:
Ahh. Right. These PTEs are zero initialized by the bootloader, so Owner will be 0 aka kernel mode.